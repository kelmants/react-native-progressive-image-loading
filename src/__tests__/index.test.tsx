import React from 'react';
import { render } from '@testing-library/react-native';
import ProgressiveImage from '../';

test('<ProgressiveImage> Component with default values', async () => {
  const { toJSON } = render(<ProgressiveImage style={{ width: 200, height: 200, marginVertical: 20 }} source={{ uri: 'https://images.pexels.com/photos/6044441/pexels-photo-6044441.jpeg' }} />);
  expect(toJSON()).toMatchSnapshot();
});

test('<ProgressiveImage> Component with thumbnailSource', async () => {
  const { toJSON } = render(
    <ProgressiveImage
      style={{ width: 200, height: 200, marginVertical: 20 }}
      source={{ uri: 'https://images.pexels.com/photos/6044441/pexels-photo-6044441.jpeg' }}
      thumbnailSource={{ uri: 'https://images.pexels.com/photos/6044441/pexels-photo-6044441.jpeg' }}
    />
  );
  expect(toJSON()).toMatchSnapshot();
});
