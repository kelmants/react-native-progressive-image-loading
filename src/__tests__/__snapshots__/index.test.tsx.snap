// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressiveImage> Component with default values 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e1e4e8",
    }
  }
>
  <Image
    onLoad={[Function]}
    source={
      Object {
        "uri": "https://images.pexels.com/photos/6044441/pexels-photo-6044441.jpeg",
      }
    }
    style={
      Object {
        "bottom": 0,
        "height": 200,
        "left": 0,
        "marginVertical": 20,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 200,
      }
    }
  />
</View>
`;

exports[`<ProgressiveImage> Component with thumbnailSource 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e1e4e8",
    }
  }
>
  <Image
    blurRadius={1}
    onLoad={[Function]}
    resizeMethod="scale"
    source={
      Object {
        "uri": "https://images.pexels.com/photos/6044441/pexels-photo-6044441.jpeg",
      }
    }
    style={
      Object {
        "height": 200,
        "marginVertical": 20,
        "opacity": 0,
        "width": 200,
      }
    }
  />
  <Image
    onLoad={[Function]}
    resizeMethod="scale"
    source={
      Object {
        "uri": "https://images.pexels.com/photos/6044441/pexels-photo-6044441.jpeg",
      }
    }
    style={
      Object {
        "bottom": 0,
        "height": 200,
        "left": 0,
        "marginVertical": 20,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 200,
      }
    }
  />
</View>
`;
